src/Entities/Tetrahedrons.jl

```diff
@@ -89,57 +96,55 @@ function _jacobian_mat(tetrahedron_coords_matrix::AbstractMatrix, derivatives::A
     tetrahedron_coords_matrix * derivatives'
 end
 
-"Computes volume element of a tetrahedron given J = det(𝔽)."
+"Computes volume element of a tetrahedron given J = det(F)."
 function _volume(jacobian_mat::AbstractMatrix)
     volume = det(jacobian_mat) / 6.0
     @assert volume > 0 throw(ArgumentError("Element with negative volume, check connectivity."))
     volume
 end
 
-function _B_mat(deriv::AbstractMatrix, 𝔽::AbstractMatrix)
+function _B_mat(deriv::AbstractMatrix, F::AbstractMatrix)
     B = zeros(6, 12)
 
-    B[1:3, :] = [diagm(deriv[:, 1]) * 𝔽' diagm(deriv[:, 2]) * 𝔽' diagm(deriv[:, 3]) * 𝔽' diagm(deriv[:,
+    B[1:3, :] = [diagm(deriv[:, 1]) * F' diagm(deriv[:, 2]) * F' diagm(deriv[:, 3]) * F' diagm(deriv[:,
                                                                                                      4]) *
-                                                                                         𝔽']
+                                                                                         F']
 
     for k in 1:4
-        B[4:6, (k - 1) * 3 .+ (1:3)] = [deriv[2, k] * 𝔽[:, 3]' + deriv[3, k] * 𝔽[:, 2]'
-                                        deriv[1, k] * 𝔽[:, 3]' + deriv[3, k] * 𝔽[:, 1]'
-                                        deriv[1, k] * 𝔽[:, 2]' + deriv[2, k] * 𝔽[:, 1]']
+        B[4:6, (k - 1) * 3 .+ (1:3)] = [deriv[2, k] * F[:, 3]' + deriv[3, k] * F[:, 2]'
+                                        deriv[1, k] * F[:, 3]' + deriv[3, k] * F[:, 1]'
+                                        deriv[1, k] * F[:, 2]' + deriv[2, k] * F[:, 1]']
     end
     B
 end
 
-function internal_forces(m::AbstractHyperElasticMaterial, t::Tetrahedron, u_e::AbstractVector)
+"Return the internal forces of a `Tetrahedron` element `t` doted with and `AbstractMaterial` `m` and
+an element displacement vector `u_e`. "
+function internal_forces(m::AbstractMaterial, t::Tetrahedron, u_e::AbstractVector)
     internal_forces(m, t, u_e, TetrahedronCache())
 end
 
 "Return the internal force of a `Tetrahedron` element `t` doted with an `AbstractHyperElasticMaterial` `m` +
-and a an element displacement vector `u_e`."
+and a an element displacement vector `u_e`. This function modifies the cache to avoid memory allocations."
 function internal_forces(m::AbstractHyperElasticMaterial, t::Tetrahedron, u_e::AbstractVector,
                          cache::TetrahedronCache)
-    (; fint, Ks, σ, ε, F, H) = cache
+    (; fint, Ks, σ, ε, F, H, X, J, funder) = cache
 
+    # Kinematics
     ∂X∂ζ = _shape_functions_derivatives(t)
-    X = _coordinates_matrix(t)
+    X .= _coordinates_matrix(t)
```

Comment:
This seems like something that can be avoided altogether, by using matrix-vector operations when computing `J` below I guess.

Reply:
Can you give me draft ? Do you mean to wrap all this code:
```julia
    ∂X∂ζ = _shape_functions_derivatives(t)
    X .= _coordinates_matrix(t)
    J .= _jacobian_mat(X, ∂X∂ζ)
    vol = _volume(J)
    funder .= inv(J)' * ∂X∂ζ
    ```
    in a single function ?

Reply:
Can you explain this ?